File: Colivi-backend/src/main/java/com/vvu981/colivibackend/features/report/service/ReportService.java
package com.vvu981.colivibackend.features.report.service;

import com.vvu981.colivibackend.features.report.dto.CreateReportRequest;
import com.vvu981.colivibackend.features.report.dto.ReportFeedbackResponse;
import com.vvu981.colivibackend.features.report.dto.ReportResponse;

import java.util.List;
import java.util.UUID;

public interface ReportService {

    ReportResponse createReport(UUID reporterId, CreateReportRequest request);

    List<ReportFeedbackResponse> getPendingFeedback(UUID reporterId);

    void acknowledgeFeedback(UUID reporterId, UUID reportId);
}

File: Colivi-frontend/src/features/report/services/reportService.ts
import api from '../../../lib/api';
import type {
  CreateReportRequest,
  ReportResponse,
  ReportFeedbackResponse,
} from '../types/report.types';

export const reportService = {
  /**
   * Submits a new report for an entity (listing, user).
   */
  createReport: async (payload: CreateReportRequest): Promise<ReportResponse> => {
    const { data } = await api.post<ReportResponse>('/reports', payload);
    return data;
  },

  /**
   * Retrieves pending resolution feedback/acknowledgments for the current user's submitted reports.
   */
  getPendingFeedback: async (): Promise<ReportFeedbackResponse[]> => {
    const { data } = await api.get<ReportFeedbackResponse[]>('/reports/pending-feedback');
    return data;
  },

  /**
   * Marks a report feedback notification as acknowledged/seen by the reporter.
   */
  acknowledgeFeedback: async (reportId: string): Promise<void> => {
    await api.patch(`/reports/${reportId}/acknowledge-feedback`);
  },
};

